internal/appconfig/machines.go

```diff
@@ -190,6 +190,12 @@ func (c *Config) updateMachineConfig(src *api.MachineConfig) (*api.MachineConfig
 	// Files
 	machine.MergeFiles(mConfig, c.MergedFiles)
 
+	// Restart Policy
+	mConfig.Restart = api.MachineRestart{
+		Policy:     api.MachineRestartPolicy(c.Restart.Policy),
+		MaxRetries: c.Restart.MaxRetries,
+	}
```

Comment:
that this block is not guarded with `if c.Restart != nil {` suggests it needs a test case. 

Follow the pattern of the other fields, nullify them and then set if set in appconfig.

```
mConfig.Restart = nil
if c.Restart != nil {
  mConfig.Restart = ...
}
```